smithy-openapi/src/main/java/software/amazon/smithy/openapi/fromsmithy/protocols/AbstractRestProtocol.java

```diff
@@ -437,23 +453,105 @@ private void updateResponsesMapWithResponseStatusAndObject(
         Shape operationOrError = shape.hasTrait(ErrorTrait.class) ? shape : operation;
         String statusCode = context.getOpenApiProtocol().getOperationResponseStatusCode(context, operationOrError);
         ResponseObject response = createResponse(
-                context, bindingIndex, eventStreamIndex, statusCode, operationOrError);
+                context, bindingIndex, eventStreamIndex, statusCode, operationOrError, false);
+
         responses.put(statusCode, response);
     }
 
+    private void updateResponsesMapWithResponseStatusAndMultipleObjects(
+            Context<T> context,
+            HttpBindingIndex bindingIndex,
+            EventStreamIndex eventStreamIndex,
+            List<StructureShape> shapes,
+            Map<String, ResponseObject> responses
+    ) {
+        List<ResponseObject> errors = new ArrayList<>();
+        ResponseObject.Builder combinedResponse = ResponseObject.builder();
+        String statusCode = context.getOpenApiProtocol().getOperationResponseStatusCode(context, shapes.get(0));
+        OpenApiConfig.ErrorStatusConflictHandlingStrategy strategy = context.getConfig().getOnErrorStatusConflict();
+        Map<String, List<MediaTypeObject>> contents = new LinkedHashMap<>();
+
+        // make response object for each error response & add to list
+        for (StructureShape shape : shapes) {
+            errors.add(createResponse(context, bindingIndex, eventStreamIndex, statusCode, shape,
+                    true));
+        }
+
+        // combine the error response objects into one response object
+        for (ResponseObject error : errors) {
+            // combine headers
+            for (Map.Entry<String, Ref<ParameterObject>> header : error.getHeaders().entrySet()) {
+                combinedResponse.putHeader(header.getKey(), header.getValue());
+            }
+
+            // get mapping of contents
+            for (Map.Entry<String, MediaTypeObject> content : error.getContent().entrySet()) {
+                contents.computeIfAbsent(content.getKey(), (k) -> new ArrayList<>()).add(content.getValue());
+            }
+
+            // combine links
+            for (Map.Entry<String, Ref<LinkObject>> link : error.getLinks().entrySet()) {
+                combinedResponse.putLink(link.getKey(), link.getValue());
+            }
+        }
```

Comment:
You can also use streams for this, e.g.,

```
        Map<String, List<MediaTypeObject>> contents =
                errors.stream()
                        .peek(error -> error.getHeaders().forEach(combinedResponse::putHeader))
                        .peek(error -> error.getLinks().forEach(combinedResponse::putLink))
                        .map(ResponseObject::getContent)
                        .map(Map::entrySet)
                        .flatMap(Collection::stream)
                        .collect(Collectors.groupingBy(Map.Entry::getKey,
                                LinkedHashMap::new,
                                Collectors.mapping(Map.Entry::getValue, Collectors.toList())));

```

Reply:
the for loop is more readable IMO

Reply:
for-loop kept, following discussion.